### PR TITLE
Create slider homepage proposals

### PR DESCRIPTION
**Pull request by eliegaboriau**
This PR creates the slider proposals content blocks 

Notion card : https://www.notion.so/opensourcepolitics/bd4bf860eab94f9daca7da803f09e3b2?v=c96df65755dd403d95e86f82d7749709&p=b10b8e8bbf9f4641a6c6b4f26a6e4b56&pm=c

### CODE REVIEW

File: app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Decidim
+  module HomepageProposals
+    module ContentBlocks
+      class ProposalsSliderCell < Decidim::ViewModel
+        attr_accessor :glanced_proposals
+
+        include Cell::ViewModel::Partial
```
Comment (Quentinchampenois) [inline app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb:9]: I guess there is unused includes and methods in this cell, we should cleanup this file and keep only what we need

File: app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        def content_block_settings
+          @content_block_settings ||= Decidim::ContentBlock.find_by(
+            manifest_name: "proposals_slider",
+            organization: current_organization
+          ).settings
+        end
+
+        def options_for_default_component
+          components = Decidim::Component.where(id: content_block_settings.linked_components_id.reject(&:blank?).map(&:to_i))
```
Comment (Quentinchampenois) [inline app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb]:
```suggestion
          components = Decidim::Component.where(id: content_block_settings.linked_components_id.compact.map(&:to_i))
```

`compact` is equivalent to `reject(&:blank?)`.

File: app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          options = components.map do |component|
+            ["#{translated_attribute(component.name)} (#{translated_attribute(component.participatory_space.title)})", component.id]
+          end
+
+          return options_for_select(options, selected: params[:filter][:component_id]) if params[:filter].present? && params[:filter][:component_id].present?
```
Comment (Quentinchampenois) [inline app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb]:
```suggestion
          return options_for_select(options, selected: params.dig(:filter, :component_id) if params.dig(:filter, :component_id).present?
```

Comment (Quentinchampenois) [reply app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb]: I just saw similar return clause just after, we should create private method which returns the selected item, the params component_id or the default_linked_component

File: app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+          options_for_select(options, selected: content_block_settings.default_linked_component)
+        end
+
+        def linked_components
+          @linked_components ||= Decidim::Component.where(id: content_block_settings.linked_components_id.reject(&:blank?).map(&:to_i))
```
Comment (Quentinchampenois) [inline app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb]:
```suggestion
          @linked_components ||= Decidim::Component.where(id: content_block_settings.linked_components_id.compact.map(&:to_i))
```

To test:  If the `content_block_settings.linked_components_id.compact` returns an Array of ID as String, we probably have not to use `.map(&:to_i)`

File: app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        end
+
+        def default_filter_params
```
Comment (Quentinchampenois) [inline app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb:38]: I think `default_filter_params` is not used in Cell

File: app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_settings_form_cell.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+        def options_for_proposals_components
+          options = proposals_components.map do |proposal_component|
+            ["#{translated_attribute(proposal_component.name)} (#{translated_attribute(proposal_component.participatory_space.title)})", proposal_component.id]
+          end
+          options_for_select(options, selected: content_block.settings.linked_components_id)
+        end
+
+        def options_for_default_component
+          components = Decidim::Component.where(id: content_block.settings.linked_components_id.reject(&:blank?).map(&:to_i))
```
Comment (Quentinchampenois) [inline app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_settings_form_cell.rb]:
```suggestion
          components = Decidim::Component.where(id: content_block.settings.linked_components_id.compact)
```

File: app/controllers/decidim/proposals_slider_controller.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+          url: proposal_path(proposal),
+          image: image_for(proposal),
+          state: proposal.state
+        }
+      end
+    end
+
+    def glanced_proposals
+      if params[:filter].present?
+        category = Decidim::Category.find(params[:filter][:category_id]) if params[:filter][:category_id].present?
```
Comment (Quentinchampenois) [inline app/controllers/decidim/proposals_slider_controller.rb]:
```suggestion
        params.dig(:filter, :category_id) if params.dig(:filter, :category_id).present?
```

File: app/controllers/decidim/proposals_slider_controller.rb [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        scopes = Decidim::Scope.find(params[:filter][:scope_id]) if params[:filter][:scope_id].present?
```
Comment (Quentinchampenois) [inline app/controllers/decidim/proposals_slider_controller.rb]:
```suggestion
       params.dig(:filter, :scope_id) if params.dig(:filter, :scope_id).present?
```

File: app/controllers/decidim/proposals_slider_controller.rb [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+      end
+
+      @glanced_proposals ||= Decidim::Proposals::Proposal.published
+                                                         .where(component: params[:filter][:component_id])
```
Comment (Quentinchampenois) [inline app/controllers/decidim/proposals_slider_controller.rb]:
```suggestion
 .where(component: params.dig(:filter, :component_id)
```

File: app/packs/src/decidim/homepage_proposals/glidejs/glideItems/Proposal.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+export default class Proposal extends GlideItem {
+    constructor(obj) {
+        super();
+        this.title = obj.title;
+        this.body = obj.body;
+        this.image = obj.image;
+        this.url = obj.url;
+        this.state = obj.state || 'not answered' ;
+        switch(this.state){
```
Comment (Quentinchampenois) [inline app/packs/src/decidim/homepage_proposals/glidejs/glideItems/Proposal.js]: Move switch case in a dedicated method

**Review by Quentinchampenois — CHANGES_REQUESTED** [review]
We also need to : 
- Create rake task to load JS dependencie `GlideJS`
- Change ProposalNotFound visit link in template to the default_component url
- Add release CI (maybe in another PR)

File: app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb (hunk 5 above)
Comment (eliegaboriau) [reply app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb:38]: It is, it crashes without

**Review by AyakorK — APPROVED** [review]
LGTM !
 Just a few questions : 
 - Shouldn't we hide the slider if he's activated unless there is a component selected ?
 - Couldn't we add an "auto-scroll" if this is possible to navigate through more proposals ?
 - Should we rework a bit the css of the filters (Actually clear filters is next to the filters which renders something weird)
 - It looks like it is by default activated as a component of the homepage, should we try to change that ?

**Review by Quentinchampenois — APPROVED** [review]

**Pull request merged**